packages/plugin-ma-tracking/src/ma-tracking.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  AdditionalBrowserOptions,
+  BaseEvent,
+  BeforePlugin,
+  BrowserConfig,
+  Campaign,
+  Event,
+  PluginType,
+  Storage,
+} from '@amplitude/analytics-types';
+import type { AmplitudeBrowser } from '@amplitude/analytics-browser';
+import { AdvancedCampaignTracker } from './advanced-campaign-tracker';
+
+export class MATrackingPlugin implements BeforePlugin {
+  name = 'MATracking';
+  type = PluginType.BEFORE as const;
+
+  constructor(private readonly instance: AmplitudeBrowser, private readonly options: AdditionalBrowserOptions = {}) {}
+
+  async setup(config: BrowserConfig): Promise<undefined> {
+    const attributionConfig = this.options.attribution ?? {};
+    const pageTrackingConfig =
+      this.options.trackPageViews ??
+      (attributionConfig.trackPageViews && {
+        filter: 'onAttribution',
+      });
+
+    const instance = this.instance;
+    // Share cookie storage with user session storage
+    const storage = config.cookieStorage as unknown as Storage<Campaign>;
+
+    const campaignTracker = new AdvancedCampaignTracker(config.apiKey, {
+      ...attributionConfig,
+      trackPageViews: false,
+      track: /* istanbul ignore next */ () => Promise.resolve(),
+      onNewCampaign: /* istanbul ignore next */ () => () => undefined,
+      storage,
+    });
+
+    // Disable "runAttributionStrategy" function from instance
+    instance.runAttributionStrategy = () => Promise.resolve();
+
+    // Web Attribution tracking
+    if (!attributionConfig.disabled) {
+      void campaignTracker.onPageChange(async ({ isNewCampaign, currentCampaign }) => {
+        if (isNewCampaign) {
+          if (attributionConfig.resetSessionOnNewCampaign) {
+            instance.setSessionId(Date.now());
+          }
+          await instance.track(campaignTracker.createCampaignEvent(currentCampaign));
+        }
+      });
+    }
+
+    // Page View tracking
+    if (pageTrackingConfig) {
+      const filter = typeof pageTrackingConfig === 'boolean' || pageTrackingConfig.filter;
```

Comment:
imo filter type being `boolean` and `{ filter: ... }` is a little hard to follow. do we need the boolean value or can we just have a default value for the object and its keys?

Reply:
Given that page tracking is opt in and has no required options, it seems weird to pass `trackPageViews: {}` to enable it rather than supporting `true`

Reply:
Since this tracking is now in a plugin, we could break the page view tracking options out from the on/off toggle pretty easily, ie. have `trackPageViews` and `pageViewTrackingOptions` as separate plugin configuration keys.

Reply:
`trackPageViews` is valid for `analytics-ma-browser` package. For the plugin, it does not take boolean value as an option.